src/main/java/com/greghaskins/spectrum/internal/junit/TimeoutWrapper.java

```diff
@@ -0,0 +1,41 @@
+package com.greghaskins.spectrum.internal.junit;
+
+import static com.greghaskins.spectrum.internal.hooks.NonReportingHook.nonReportingHookFrom;
+import static com.greghaskins.spectrum.internal.junit.RuleContext.statementOf;
+
+import com.greghaskins.spectrum.internal.hooks.NonReportingHook;
+
+import org.junit.internal.runners.statements.FailOnTimeout;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Wrap JUnit's timeout mechanism as a {@link NonReportingHook}.
+ */
+public interface TimeoutWrapper {
```

Comment:
It kind of made sense to have the `BlockTimeout` not do its own implementation of timeout and to move our dependency on JUnit4 into the `junit` package. This means we could implement the timeout hook another way in future and only change the factory method used... probably a sensible bit of design.